search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,170 @@
+		/**
+		 * function: initial_solution
+		 * Initialize a new solution, generating
+		 * random numbers with the maximum height of the board.
+		 * @param [in] nqueens is the number of queens to be placed on the chessboard.
+		 * The number of queens is also the dimension of the chessboard
+		 * @returns a vector that will be the starting solution.
+		 */
+		std::vector<int> initial_solution(int nqueens) {
+			std::vector<int> solution;
+			solution.resize(nqueens);
+			for (auto &element : solution) {
+				element = rand() % nqueens + 1;
+			}
+			return solution;
+		}
+
+		/**
+		 * function: cost
+		 * Calculate the number of pairs of queens that are attacked within the board,
+		 * the optimal cost is h = 1. Read the 8 queens problem.
+		 * @param [in] solution_vector is a vector with the
+		 * column number where the queens are located on the board.
+		 * @returns the cost of the solution vector.
+		 */
+		int cost(const std::vector<int> &solution_vector) {
+			int h = 0;
+			//sum column
+			for (size_t i = 0; i < solution_vector.size(); i++) {
+				for (size_t j = i + 1; j < solution_vector.size(); j++) {
+					if (solution_vector[i] == solution_vector[j]) {
+						h++;
+						break;
+					}
+				}
+			}
+			//sum diagonal
+			for (size_t i = 0; i < solution_vector.size(); i++) {
+				for (size_t j = i + 1; j < solution_vector.size(); j++) {
+					if ((solution_vector[i] - (i + 1))
+							== (solution_vector[j] - (j + 1))) {
+						h++;
+						break;
+					}
+				}
+			}
+			//sum inverse diagonal
+			for (size_t i = 0; i < solution_vector.size(); i++) {
+				for (size_t j = i + 1; j < solution_vector.size(); j++) {
+					if ((solution_vector[i] + (i + 1))
+							== (solution_vector[j] + (j + 1))) {
+						h++;
+						break;
+					}
+				}
+			}
+			return h;
+		}
+
+		/**
+		 * function: neighbour
+		 * Generate a neighbor based on the current solution of
+		 * the problem. Read about neighborhoods in local search.
+		 * @param [in] solution_vector is a vector with the
+		 * column number where the queens are located on the board.
+		 * @returns a new vector similar to the one passed by parameter.
+		 */
+		std::vector<int> neighbour(std::vector<int> solution_vector) {
+			int64_t i = rand() % solution_vector.size() + 1;
+			int64_t mov = (rand() % solution_vector.size() + 1);
+			solution_vector[i - 1] = mov;
+			return solution_vector;
+		}
+
+		/**
+		 * function: print_solution
+		 * Prints the solution obtained on the chessboard in console.
+		 * @param [in] solution_vector is a vector with the
+		 * column number where the queens are located on the board.
+		 */
+		void print_solution(const std::vector<int> &solution_vector) {
+			for (int i = 0; i < solution_vector.size(); i++) {
+				for (int j = 0; j < solution_vector.size(); j++) {
+					if (j == solution_vector[i] - 1) {
+						std::cout << " Q ";
+					} else {
+						std::cout << " _ ";
+					}
+				}
+				std::cout << "\n";
+			}
+
+			std::cout << "\nSolucion: ";

```

Comment:
Please consider using the English language, rather than any other language.
```suggestion
			std::cout << "\nSolution: ";

```